rules/collapsed-subprocess.js

```diff
@@ -0,0 +1,36 @@
+const { is } = require('bpmnlint-utils');
+
+const { ERROR_TYPES } = require('./utils/element');
+
+const { reportErrors } = require('./utils/reporter');
+
+module.exports = function() {
+  function check(di, reporter) {
+
+    if (!isCollapsedSubProcess(di)) {
+      return;
+    }
+
+    const node = di.bpmnElement;
+
+    const error = {
+      message: `A <${ node.$type }> must be expanded.`,
+      error: {
+        type: ERROR_TYPES.ELEMENT_COLLAPSED_NOT_ALLOWED,
+        node: node,
+        parentNode: null
+      }
+    };
+
+    reportErrors(node, reporter, error);
+  }
+
+  return {
+    check
+  };
+};
+
+
+function isCollapsedSubProcess(di) {
+  return is(di, 'bpmndi:BPMNPlane') && is(di.bpmnElement, 'bpmn:SubProcess');
+}
```

Comment:
This relies heavily on our behavior that BPMNDiagrams are created automatically for collapsed subprocesses. However, if I import diagram from an external source (or use a legacy diagram), it does not need to have a separate BPMNDiagram for each collapsed subprocess. I'd suggest to rather check the element type and it's DI's `isExpanded === false` property.

Reply:
As in https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/52/files#r1011454619

Reply:
> Great catch! I adjusted the PR with your suggestion

Shall we have a test case for this?